### PR TITLE
Support $GOPATH imports

### PR DESCRIPTION
**Pull request by ahawkins**
This commits puts the code into $GOPATH/src/$REPO so package specific
imports work. This is required since gometalinter parses code to detect
problems.

This poses a new question. What about dependencies not available in source control. Should TeamCI run `go get` or the equivalent to download dependencies?

### CODE REVIEW

File: gometalinter/gometalinter-tap [hunk 1]
```diff
@@ -23,12 +23,19 @@ main() {
 		gometalinter_args+=(--config "${config_file}")
 	fi
 
+	echo '~~~ Preparing code for $GOPATH'
+	mkdir -p "${GOPATH}/src/github.com/${TEAMCI_REPO_SLUG}"
+	cp -vr . "${GOPATH}/src/github.com/${TEAMCI_REPO_SLUG}"
```
Comment (ahawkins) [inline gometalinter/gometalinter-tap]: @antonlindstrom Is it safe to assume that most projects are developed in this directory structure? I mean, your development work happens inside the `$GOPATH/src/github.com/mycompany/repo`directory.

Comment (antonlindstrom) [reply gometalinter/gometalinter-tap]: I think this will cover 99% of the cases :+1:

**Comment by antonlindstrom** [comment]
@ahawkins Regarding the dependencies this is a hard one since there are multiple ways to getting dependencies. There are certain projects that uses `dep`, some that uses `vgo` and some that just use `go get`. 

If it is possible to lookup which tool is being used and run that command, that would be really nice.

**Comment by ahawkins** [comment]
How would lookup work? Also what about private dependencies? 

// Adam

**Comment by antonlindstrom** [comment]
Good question about private dependencies, not sure what to do about that at this point. I'm thinking that the lookup would be, for `dep` that a `Gopkg.toml` exists and for `vgo` that a `go.mod` file exists. Maybe using `go get ./...` may be enough as a fallback?

**Pull request merged**